### DISCUSSION TITLE
Null value for KafkaProducerSampler.java class and unable to load jar in Jmeter

### DISCUSSION THREAD

**Discussion by dhananjay1308** [sngular/kloadgen#344] [Q&A] [answered]
I have created original-kloadgen-5.3.1.jar from the steps given in readme but when I checked in decompiler com.sngular.kloadgen.sampler.KafkaProducerSampler.java is having null value.
Please check attachment (original-kloadgen-5.3.1.txt) please change extension from txt to jar .

Also from the 5th step given on https://github.com/sngular/kloadgen/wiki/how-to-run#how-to-run-a-test-plan I have restarted jmeter after placing jar file under JMETER_HOME/lib/ext directory but still can't see com.sngular.kloadgen.sampler.KafkaProducerSampler under the dropdown.

Can someone please re-verify the given steps are correct?

[original-kloadgen-5.3.1.txt](https://github.com/sngular/kloadgen/files/10849583/original-kloadgen-5.3.1.txt)

**Comment by josemenvegsngular** [comment, marked as answer]
Hi, @dhananjay1308 ,

Reviewing what you say, I think that you have copied the wrong file.

You can review it under [installation](https://github.com/sngular/kloadgen/wiki/installation), it says:
`Once the installation is complete, copy target/kloadgen-<version>.jar file to the JMETER_HOME/lib/ext directory.`
The previous step create two files:

- **kloadgen-5.3.1.jar**: 49,7 MB (the correct one)
- **original-kloadgen-5.3.1.jar**: 299,6 kB the one that you have copied

Please, copy the first one, and tell us if it solve your problem (or not).

**Reply by dhananjay1308** [reply]
Hi @josemenvegsngular 

Sorry for uploading wrong file here , Yes the correct file generated was 49.7 MB. I placed it under JMETER_HOME/lib/ext but still I was not able to locate com.sngular.kloadgen.sampler.KafkaProducerSampler as mentioned in step#5 here -->https://github.com/sngular/kloadgen/wiki/how-to-run#how-to-run-a-test-plan.

After updating JMETER_HOME/bin/user.properties with required classpath - 
user.classpath=/Users/dhananjayshinde/Documents/apache-jmeter-5.1.1/lib/ext/kloadgen-5.3.1.jar it was loaded.

So just in case someone faces similar issue try updating user.properties with required classpath.

Thanks I will keep updating if I face any further issues.

**Reply by josemenvegsngular** [reply]
Hi @dhananjay1308 

So you say that your doesn't see something like this:
![image](https://user-images.githubusercontent.com/125871857/221870345-78128408-c422-4bff-ab76-1aa11acc3d73.png) or it has been solved when you have updated the properties.

I haven't got the properties file modified (may be it's an OS issue, I'm in a linux).

Or that I'm calling jmeter.sh from it's own folder (and I don't need to update the path).

**Reply by dhananjay1308** [reply]
Hi @josemenvegsngular 

Yes I saw this only  after updating user.properties explicitly not sure may be some version compatibility issue, Thanks a lot for your quick help here.
